bin/-tarball-info.js

```diff
@@ -166,7 +166,9 @@ function generatePackageReference(version, tarballName) {
 }
 
 function insertTarballsToPackageJson(fileLocation, options = {}) {
-  const pkgInfo = require(fileLocation);
+  const location = require.resolve(fileLocation);
+  delete require[location];
```

Comment:
Why do we need to do this? If the intent is to purge the require cache, this isn't the correct way to do it.

The simplest thing would be to do `delete require.cache[location]`, but you may still leak the prior entry in the specific parent that required it.

See https://github.com/ember-cli/ember-cli-htmlbars/blob/d1b9683dc770042be91f4ac96f93d58dcaaed74f/lib/utils.js#L113-L126 for a reference.

Reply:
I forgot the .cache thanks for catching that

Reply:
As for why, this is mostly a preventative measure since for the data packages we sometimes write a version to the file earlier in the process in some command flows.